pkg/util/clusteragent/clcrunner.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package clusteragent
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+/*
+Client to query the Datadog Cluster Level Check Runner API.
+*/
+
+const (
+	clcRunnerPath        = "api/v1/clcrunner"
+	clcRunnerVersionPath = "version"
+	clcRunnerStatsPath   = "stats"
+)
+
+var globalCLCRunnerClient *CLCRunnerClient
+
+// CLCRunnerClientInterface is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClientInterface interface {
+	GetVersion(IP string) (version.Version, error)
+	GetRunnerStats(IP string) (types.CLCRunnersStats, error)
+}
+
+// CLCRunnerClient is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClient struct {
+	clcRunnerAPIRequestHeaders http.Header
+	clcRunnerAPIClient         *http.Client
+	clcRunnerPort              int
+}
+
+// resetGlobalCLCRunnerClient is a helper to remove the current CLCRunnerClient global
+// It is ONLY to be used for tests
+func resetGlobalCLCRunnerClient() {
+	globalCLCRunnerClient = &CLCRunnerClient{}
+	globalCLCRunnerClient.init()
+}
+
+// GetCLCRunnerClient returns or init the CLCRunnerClient
+func GetCLCRunnerClient() (CLCRunnerClientInterface, error) {
```

Comment:
As suggested before if you use `sync.Once`

```suggestion
func GetCLCRunnerClient() (CLCRunnerClientInterface, error) {
      globalCLCRunnerClient.Do(globalCLCRunnerClient.init())
      return globalCLCRunnerClient, globalCLCRunnerClient.initErr
}
```